test/archive.cpp

```diff
@@ -280,6 +280,57 @@ TEST(ZimArchive, openSplitZimArchive)
   }
 }
 
+struct TestCacheConfig {
+  size_t dirent_cache_size;
+  size_t cluster_cache_size;
+  size_t dirent_lookup_cache_size;
+};
+
+TEST(ZimArchive, cacheDontImpactReading)
+{
+  const TestCacheConfig cache_configs[] = {
+    {0, 0, 0},
+    {1, 1, 1},
+    {2, 2, 2},
+    {10, 10, 10},
+    {1000, 2000, 1000},
+    {0, 2000, 1000},
+    {1000, 0, 1000},
+    {1000, 2000, 0},
+    {1, 2000, 1000},
+    {1000, 1, 1000},
+    {1000, 2000, 1},
+  };
+
+  for (auto& testfile: getDataFilePath("small.zim")) {
+    auto ref_archive = zim::Archive(testfile.path);
+
+    for (auto cache_config: cache_configs) {
+      auto test_archive = zim::Archive(testfile.path);
+      test_archive.set_dirent_cache_max_size(cache_config.dirent_cache_size);
+      test_archive.set_dirent_lookup_cache_max_size(cache_config.dirent_lookup_cache_size);
+      test_archive.set_cluster_cache_max_size(cache_config.cluster_cache_size);
+
+      auto ref_range = ref_archive.iterEfficient();
+      for (auto ref_entry:ref_range) {
+        auto test_entry = test_archive.getEntryByPath(ref_entry.getPath());
+        EXPECT_EQ(ref_entry.getPath(), test_entry.getPath());
+        EXPECT_EQ(ref_entry.getTitle(), test_entry.getTitle());
+        EXPECT_EQ(ref_entry.isRedirect(), test_entry.isRedirect());
+        if (ref_entry.isRedirect()) {
+          EXPECT_EQ(ref_entry.getRedirectEntryIndex(), test_entry.getRedirectEntryIndex());
+        } else {
+          auto ref_item = ref_entry.getItem();
+          auto test_item = test_entry.getItem();
+          EXPECT_EQ(ref_item.getClusterIndex(), test_item.getClusterIndex());
+          EXPECT_EQ(ref_item.getBlobIndex(), test_item.getBlobIndex());
+          EXPECT_EQ(ref_item.getData(), test_item.getData());
```

Comment:
Isn't it better to use `ASSERT_EQ()` instead of `EXPECT_EQ()`? Otherwise in case of a bug in caching there may be a lot of output with little value.